nexis/lib/widgets/auth/custom_column.dart

```diff
@@ -0,0 +1,192 @@
+  final String? mediumBody;
+  final String? smallLabel;
+  final TextEditingController? controller;
+  final void Function(String)? onSubmitted;
+  final GestureRecognizer? recognizer;
+  final bool? obscureText;
+  final String? hint;
+  final void Function()? onTap;
+  final void Function()? onPressed;
+  final String? buttonText;
+  final TextEditingController? monthController;
+  final TextEditingController? dayController;
+  final TextEditingController? yearController;
+  final bool? readOnly;
+
+  const CustomColumn(
+    this.type,
+    {this.largeLabel,
+    this.mediumLabel,
+    this.mediumBody,
+    this.smallLabel,
+    this.controller,
+    this.onSubmitted,
+    this.recognizer,
+    this.obscureText,
+    this.hint,
+    this.onTap,
+    this.onPressed,
+    this.buttonText,
+    this.monthController,
+    this.dayController,
+    this.yearController,
+    this.readOnly,
+    super.key
+    });
+
+  @override
+  Widget build(BuildContext context) {
+    List<double> margins = getMargins(type);
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        heading(type, context, largeLabel),
+        sizedBox(margins[0]),
+        text(type, context, mediumLabel),
+        sizedBox(margins[1]),
+        textField(type, controller, onSubmitted, obscureText, hint, onTap, readOnly),
+        rowCol(type, context, onTap, monthController, dayController, yearController),
+        sizedBox(margins[2]),
+        richText(type, context, mediumBody, recognizer),
+        button(type, onPressed, buttonText),
+        sizedBox(margins[3]),
+        richText2(type, context, smallLabel, mediumBody, recognizer),
+      ],
+    );
+  }
+
+  heading(ColumnType type, BuildContext? context, String? largeLabel) {
+    if (type != ColumnType.type1) { return sizedBox(0); }
+    return Text(
+      largeLabel ?? "", 
+      style: Theme.of(context!).textTheme.labelLarge,
+    );
+  }
+
+  textField(ColumnType type, TextEditingController? controller, void Function(String)? onSubmitted, bool? obscureText, 
+    String? hint, void Function()? onTap, bool? readOnly) {
+    if (type == ColumnType.type4 || type == ColumnType.type5) { return sizedBox(0); }
+    return SizedBox(
+      width: 500,
+      child: CustomTextField(
+        obscureText: obscureText ?? false,
+        hint: hint ?? '',
+        controller: controller,
+        onSubmitted: onSubmitted,
+        onTap: onTap,
+        readOnly: readOnly ?? false,
+      ),
+    );
+  }
+
+  text(ColumnType type, BuildContext? context, String? mediumLabel){
+    if (type == ColumnType.type4) { return sizedBox(0); }
+    return Text(
+      mediumLabel ?? "", 
+      style: Theme.of(context!).textTheme.labelMedium,
+    );
+  }
+
+  sizedBox(double? height) {
+    return SizedBox(height: height);
+  }
+
+  richText(ColumnType type, BuildContext? context, String? mediumBody, GestureRecognizer? recognizer) {
```

Comment:
Combine these two together, since there's significant code duplication. This could be refactored to use a single method with some conditional logic.